WordPress/gutenberg
# Test: Validate block & theme json

## What?

Validate more block.json and theme.json files.

Follow up to #57201 that discovered some invalid block and theme json files.

## Why?

These JSON files should be valid according to schemas. If they're invalid, either the schema or the JSON is wrong and should be fixed.

## How?

Add and expand integration tests.

## Testing Instructions

CI passes. Run these tests locally with `npm run test:unit schema.test.js`.

You can test with this patch to confirm globs are working as expected:

```diff
diff --git a/phpunit/fixtures/block.json b/phpunit/fixtures/block.json
index f4ef1b313c..a5ef6d0d8e 100644
--- a/phpunit/fixtures/block.json
+++ b/phpunit/fixtures/block.json
@@ -12,7 +12,7 @@
 	"icon": "star",
 	"description": "Shows warning, error or success notices…",
 	"keywords": [ "alert", "message" ],
-	"textdomain": "my-plugin",
+	"textDomain": "my-plugin",
 	"attributes": {
 		"message": {
 			"type": "string",
diff --git a/phpunit/fixtures/hooked-block/block.json b/phpunit/fixtures/hooked-block/block.json
index 791bc7b0a7..314891f420 100644
--- a/phpunit/fixtures/hooked-block/block.json
+++ b/phpunit/fixtures/hooked-block/block.json
@@ -1,6 +1,5 @@
 {
 	"$schema": "https://schemas.wp.org/trunk/block.json",
-	"title": "Hooked Block",
 	"name": "tests/hooked-block",
 	"blockHooks": {
 		"tests/group-first-child": "firstChild",
```
